tests/unit/test_agw_service_user_creator.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import pwd
+import unittest
+from unittest.mock import Mock, mock_open, patch
+
+from magma_access_gateway_installer.agw_service_user_creator import (
+    AGWInstallerServiceUserCreator,
+)
+
+
+class TestAGWInstallerServiceUserCreator(unittest.TestCase):
+
+    MAGMA_USER_DETAILS = pwd.struct_passwd(
+        ("magma", "x", 1001, 1001, ",,,", "/home/magma", "/bin/bash")
+    )
+    MAGMA_USER_NOT_IN_SUDO_GROUP = b"magma : magma\n"
+    MAGMA_USER_IN_SUDO_GROUP = b"magma : magma sudo\n"
+    ETC_SUDOERS_WITH_MAGMA_USER = """Defaults	env_reset
+Defaults	mail_badpass
+Defaults	secure_path="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"
+
+# User privilege specification
+root	ALL=(ALL:ALL) ALL
+
+# Members of the admin group may gain root privileges
+%admin ALL=(ALL) ALL
+
+# Allow members of group sudo to execute any command
+%sudo	ALL=(ALL:ALL) ALL
+
+magma ALL=(ALL) NOPASSWD:ALL
+"""
+    ETC_SUDOERS_WITHOUT_MAGMA_USER = """Defaults	env_reset
+Defaults	mail_badpass
+Defaults	secure_path="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"
+
+# User privilege specification
+root	ALL=(ALL:ALL) ALL
+
+# Members of the admin group may gain root privileges
+%admin ALL=(ALL) ALL
+
+# Allow members of group sudo to execute any command
+%sudo	ALL=(ALL:ALL) ALL
+"""
+
+    def setUp(self) -> None:
+        self.agw_service_user_creator = AGWInstallerServiceUserCreator()
+
+    @patch("magma_access_gateway_installer.agw_service_user_creator.check_call")
+    def test_given_magma_user_doesnt_exist_when_create_magma_service_user_is_called_then_magma_user_is_created(  # noqa: E501
```

Comment:
```suggestion
    def test_given_magma_user_doesnt_exist_when_create_magma_user_then_magma_user_is_created(  # noqa: E501
```

Reply:
No need for the `is_called` in the when section since it's always about public methods being called.

Reply:
Same goes for other test names in this file

Reply:
Fixed.